### PR TITLE
Supporting docs on RateDetails page 

### PR DESCRIPTION
**Pull request by haworku**
## Summary
- The significant change in this PR is reworking how file upload functions on rate details. The data storage is in Formik now.
- In addition, an exploratory component called `SingleRateCert` is used which allows us to more easily handle rates as a single rate or multiple rates in the future. I expect the props on this component will adjust as we start to do rate only workflow features.

#### Related issues
https://qmacbis.atlassian.net/browse/MR-3282

#### Test cases covered
Adjusted tests to also check that supporting docs behaves as expected

## QA guidance
Double check that features on RateDetails including file uploads are working as expected

### CODE REVIEW

**Comment by haworku** [comment]
I think things are good here. There is one test I "skipped" it would be nice to fix to and get passing but I manually tested the same case and I think its find.

File: services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.test.tsx
Comment (haworku) [inline services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.test.tsx:1330]: Skipped test is here - its for save as draft button behavior. It looked fine in manual QA 🤔

File: docs/technical-design/launch-darkly-testing-approach.md
Comment (JasonLin0991) [inline docs/technical-design/launch-darkly-testing-approach.md:16]: Just a typo here.

```suggestion
  - Log into [LaunchDarkly](https://app.launchdarkly.us). To login use trussworks email (e.g. `@teamtrussworks.com`) and any password, which will redirect you to CMS SSO login.
```

File: services/app-web/src/components/Select/ProgramSelect/ProgramSelect.stories.tsx
Comment (JasonLin0991) [inline services/app-web/src/components/Select/ProgramSelect/ProgramSelect.stories.tsx:12]: This storybook component is broken. The `ProgramSelect` component uses the `useStatePrograms` hook to get state programs. That hook then uses the `useAuth` hook which errors out because that `useAuth` is a can of dependency worms that Storybook does not have.

Not sure exactly what the fix here is, but we essentially have to "mock" the `useStatePrograms` to return some test programs. Maybe spin this work off on its own ticket.

File: services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx
Comment (JasonLin0991) [inline services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx:160]: Regression: After selecting a shared rate from the dropdown, clicking continue, and then clicking back, the selected shared rate is no longer selected in the dropdown. This also occurs when saving a draft.

The selected package still exists in the data, it’s just not being displayed.

File: services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx
Comment (JasonLin0991) [inline services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx:31]: `PackagesWithSharedRates` key prop needs to be renamed maybe to `keyProp`.

```suggestion
    keyProp,
```

Screenshot 2023-06-26 at 10 37 17 PM

File: services/app-web/src/pages/StateSubmission/RateDetails/SingleRateCert/SingleRateCert.tsx
Comment (JasonLin0991) [inline services/app-web/src/pages/StateSubmission/RateDetails/SingleRateCert/SingleRateCert.tsx:294]: I noticed an error in the console when clicking on the program select validation error in the error summary.

File: services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx
Comment (JasonLin0991) [inline services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.tsx:146]: **NIT**: When adding another rate cert, the supporting documents upload section is highlighted instead of the rate cert document upload section.

File: services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx
Comment (JasonLin0991) [reply services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx:160]: I'm wondering if we should add the step of continuing to the next section of the form and then back to the Cypress test.

**Review by JasonLin0991 — CHANGES_REQUESTED** [review]
Great work 🔥. This is such a great refactor and will help whenever we try to implement rate only submission.

I wont approve yet though. The regression with the packageSelect mentioned below is something I would want to see before we merge in.

File: services/app-web/src/pages/StateSubmission/RateDetails/SingleRateCert/SingleRateCert.tsx
Comment (JasonLin0991) [reply services/app-web/src/pages/StateSubmission/RateDetails/SingleRateCert/SingleRateCert.tsx:294]: Oops! here is the rest of the info for this error
Screenshot 2023-06-27 at 10 16 59 AM

This error is from `ErrorSummary` component using `querySelector` to focus on the combo box. `querySelector` does not like special characters when passed in in the format `#rateInfos.0.rateProgramIDs`.

To fix this we just need to change the `fieldSelector` value to `[id=$:{errorKey}"]`
Screenshot 2023-06-27 at 10 21 12 AM

File: services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx
Comment (JasonLin0991) [reply services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx:31]: Fixing this prop might fix the regression in displaying selected shared rates.

File: services/app-web/src/pages/StateSubmission/RateDetails/SingleRateCert/SingleRateCert.tsx
Comment (JasonLin0991) [reply services/app-web/src/pages/StateSubmission/RateDetails/SingleRateCert/SingleRateCert.tsx:294]: @rswerve I think I might have misled you by highlighting the else block. The code that needs to be changed in on line 18. From
```typescript
 fieldSelector = errorKey
```
to 
```typescript
fieldSelector = `[id="${errorKey}"]`
```

File: services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.test.tsx
Comment (rswerve) [reply services/app-web/src/pages/StateSubmission/RateDetails/RateDetails.test.tsx]: this is working now

File: services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx
Comment (rswerve) [reply services/app-web/src/pages/StateSubmission/RateDetails/PackagesWithSharedRates/PackagesWithSharedRates.tsx:160]: made this a controlled Select, which fixed the issue.  moved the tests to cypress

**Review by JasonLin0991 — APPROVED** [review]
Just a couple of left over things but otherwise It looks great!. Thanks for updating the Cypress test.

File: services/app-web/src/components/Select/ProgramSelect/ProgramSelect.stories.tsx
Comment (haworku) [reply services/app-web/src/components/Select/ProgramSelect/ProgramSelect.stories.tsx:12]: FYI @JasonLin0991 no need to despair about dependency worms in storybook for future - you can get those dependencies in storybook - see `ProvidersDecorator`. It works very similar to what we do in tests to work around this.

**Pull request merged**
